hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java

```diff
@@ -2434,6 +2438,27 @@ String reconfigureSlowNodesParameters(final DatanodeManager datanodeManager,
     }
   }
 
+  private String reconfigureBlockInvalidateLimit(final DatanodeManager datanodeManager,
+      final String property, final String newVal) throws ReconfigurationException {
+    namesystem.writeLock();
+    try {
+      if (newVal == null) {
+        datanodeManager.setBlockInvalidateLimit(DFSConfigKeys.DFS_BLOCK_INVALIDATE_LIMIT_DEFAULT);
+        return String.valueOf(DFSConfigKeys.DFS_BLOCK_INVALIDATE_LIMIT_DEFAULT);
+      } else {
+        datanodeManager.setBlockInvalidateLimit(Integer.parseInt(newVal));
+        return String.valueOf(datanodeManager.getBlockInvalidateLimit());
+      }
```

Comment:
perhaps we should log the message here instead of in the finally block?
In addition, if newVal == null, the method returns DFS_BLOCK_INVALIDATE_LIMIT_DEFAULT, not datanodeManager.getBlockInvalidateLimit().

Reply:
Thanks, updated.